### PR TITLE
ECS: handle missing ecs exec permissions

### PR DESCRIPTION
**Pull request by gsweetwood**
Problem: 
Solution:If a user is missing the necessary ecs permission to execute
commands, a generic failed message will show when they attempt to run a
command in a container.

## Problem
If a user is missing the necessary ecs permission to execute
commands, a generic failed message will show when they attempt to run a
command in a container.
## Solution
If a user is missing the necessary ecs permission to execute
commands, a generic failed message will show when they attempt to run a
command in a container.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by gsweetwood** [comment]
Simulating assumed-role policies isn't permitted, making a pre-action check very difficult, if not impossible. I think it's best to handle the error more precisely by alerting the user why they cannot perform the action and link them to the documentation that can assist them.

File: src/ecs/commands/runCommandInContainer.ts [hunk 1]
```diff
@@ -109,14 +114,29 @@ export async function runCommandInContainer(
...
-        getLogger().error('ecs: Failed to execute command in container, %O', error)
-        showViewLogsMessage(localize('AWS.ecs.runCommandInContainer.error', 'Failed to execute command in container.'))
+        const failedMessage = localize('AWS.ecs.runCommandInContainer.error', 'Failed to execute command in container.')
+        const missingPermissions = localize(
+            'AWS.ecs.runCommandInContainer.missingPermissions',
+            'Execute Command not allowed. See documentation for required permissions.'
+        )
+        const viewDocs = localize('AWS.generic.viewDocs', 'View Documentation')
+        getLogger().error('ecs: Failed to execute command in container, %O', err)
+        if (error.name === 'AccessDeniedException') {
```
Comment (JadenSimon) [inline src/ecs/commands/runCommandInContainer.ts:131]: Just curious, but what's the `code` for this error (if any)?

File: src/ecs/commands/runCommandInContainer.ts [hunk 2]
```diff
@@ -109,14 +114,29 @@ export async function runCommandInContainer(
...
 
         result = 'Failed'
-        getLogger().error('ecs: Failed to execute command in container, %O', error)
-        showViewLogsMessage(localize('AWS.ecs.runCommandInContainer.error', 'Failed to execute command in container.'))
+        const failedMessage = localize('AWS.ecs.runCommandInContainer.error', 'Failed to execute command in container.')
+        const missingPermissions = localize(
+            'AWS.ecs.runCommandInContainer.missingPermissions',
+            'Execute Command not allowed. See documentation for required permissions.'
+        )
+        const viewDocs = localize('AWS.generic.viewDocs', 'View Documentation')
```
Comment (JadenSimon) [inline src/ecs/commands/runCommandInContainer.ts:129]: This is already in `localizedText.ts` I think.

File: src/ecs/commands/runCommandInContainer.ts [hunk 3]
```diff
@@ -109,14 +114,29 @@ export async function runCommandInContainer(
...
             return
         }
 
         result = 'Failed'
-        getLogger().error('ecs: Failed to execute command in container, %O', error)
-        showViewLogsMessage(localize('AWS.ecs.runCommandInContainer.error', 'Failed to execute command in container.'))
+        const failedMessage = localize('AWS.ecs.runCommandInContainer.error', 'Failed to execute command in container.')
+        const missingPermissions = localize(
+            'AWS.ecs.runCommandInContainer.missingPermissions',
+            'Execute Command not allowed. See documentation for required permissions.'
```
Comment (JadenSimon) [inline src/ecs/commands/runCommandInContainer.ts:127]: Wording is kind of vague. _What_ doesn't have permissions?
If it's the user's role, perhaps phrase it by placing the subject first:
`Insufficient user role permissions for "${commandName}".  See documentation for required permissions.`

The user already knows what they were wanting to do. With errors, they need to know why they **could not** perform the action.

File: src/ecs/commands/runCommandInContainer.ts (hunk 1 above)
Comment (JadenSimon) [reply src/ecs/commands/runCommandInContainer.ts:131]: They were, at one point, different in v3 (which we are not using here ofc). May have been a bug.

**Pull request merged**